src/reflection/FStar.Reflection.Basic.fs

```diff
@@ -368,7 +368,7 @@ let pack_ln (tv:term_view) : term =
             | Pat_Dot_Term (bv, t) -> wrap <| Pat_dot_term (bv, t)
         in
         let brs = List.map (function (pat, t) -> (pack_pat pat, None, t)) brs in
-        S.mk (Tm_match (t, ret_opt, brs)) Range.dummyRange
+        S.mk (Tm_match (t, ret_opt, brs, None)) Range.dummyRange
```

Comment:
I guess the additional field is hidden in the reflection API? In general, residual_comp is not exposed in reflection.